python/TestHarness/testers/Tester.py

```diff
@@ -41,6 +41,8 @@ def validParams():
     params.addParam('petsc_debug',   ['ALL'], "{False,True} -> test only runs when PETSc is configured with --with-debugging={0,1}, otherwise test always runs.")
     params.addParam('curl',          ['ALL'], "A test that runs only if CURL is detected ('ALL', 'TRUE', 'FALSE')")
     params.addParam('tbb',           ['ALL'], "A test that runs only if TBB is available ('ALL', 'TRUE', 'FALSE')")
+    params.addParam('pthreads',      ['ALL'], "A test that runs only if PTHREADS is available ('ALL', 'TRUE', 'FALSE')")
+    params.addParam('openmp',        ['ALL'], "A test that runs only if OPENMP is available ('ALL', 'TRUE', 'FALSE')")
```

Comment:
Having been bit by this myself (yes the current threads regression is my fault because I was thinking "if pthreads is enabled, I want to run this pthread join loop" - WRONG!), I'm hesitant to make these flag available at all because they are easy to misuse. I see why you think encoding the threading model is bad but making the raw flags available is worse because they can be used incorrectly if you don't fully understand the threading model order in libMesh. In the future, we are moving away from that in libMesh. Hopefully a user will be able to just choose a model and run which is why I like the enum better. I'm willing to be overruled but I think we should go ahead an encode it for now since it's unlikely to change until it's completely overhauled.